packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -1882,28 +1884,60 @@ async function runInclude(
 ): Promise<string | null> {
   const uri = await resolveIncludeFileUri(item, context);
 
-  function failToResolve(error?: any): void {
+  async function failToResolve(error?: any): Promise<void> {
     if (error) {
       console.log("Failed to resolve include file:", error);
     }
-    const diagnostic = diagnosticFromCode(
-      PLICodes.Severe.IBM3841I,
-      item.token,
-      item.fileName,
-    );
-    if (item.fileName)
-      diagnostic.data = {
-        unresolvedFile: item.fileName,
-        entryUri: context.entryUri.toString(),
-      };
+    let diagnostic: Diagnostic;
+
+    if (!(await hasConfig())) {
+      diagnostic = diagnosticFromCode(
+        Internal.DiagnosticMissingConfiguration,
+        item.token,
+      );
+      diagnostic.data = { entryUri: context.entryUri.toString() };
+    } else {
+      diagnostic = diagnosticFromCode(
+        PLICodes.Severe.IBM3841I,
+        item.token,
+        item.fileName,
+      );
+      if (item.fileName)
+        diagnostic.data = {
+          unresolvedFile: item.fileName,
+          entryUri: context.entryUri.toString(),
+        };
+    }
+
     context.diagnostics.push(diagnostic);
   }
 
   if (!uri) {
-    failToResolve();
+    await failToResolve();
     return null;
   }
 
+  async function hasConfig(): Promise<boolean> {
+    const workspaceFolderUri = URI.parse(
+      PluginConfigurationProviderInstance.getWorkspacePath(),
+    );
+    const progConfigUri = URI.parse(
+      `${workspaceFolderUri}/${PluginConfigurationProvider.PROGRAM_CONFIG_FILE}`,
+    );
+    const procsConfigUri = URI.parse(
+      `${workspaceFolderUri}/${PluginConfigurationProvider.PROCESS_GROUP_CONFIG_FILE}`,
+    );
+
+    const hasProgConfig =
+      await FileSystemProviderInstance.fileExists(progConfigUri);
+    const hasProcsConfig =
+      await FileSystemProviderInstance.fileExists(procsConfigUri);
+
+    if (!hasProcsConfig && !hasProgConfig) return false;
+
+    return true;
+  }
```

Comment:
Suggestion: The function has no dependency on any of the local variables, it can be moved out of the function.